jonason/JSONValue.hpp

```diff
@@ -0,0 +1,70 @@
+//
+//  JSONValue.hpp
+//  jonason
+//
+//  Created by Gabriel Lumbi on 2023-03-03.
+//
+
+#ifndef JSONValue_hpp
+#define JSONValue_hpp
+
+#include <unordered_map>
+#include <vector>
+#include <string>
+
+namespace jonason {
+
+struct JSONValue {
+    enum { OBJECT, ARRAY, STRING, NUMBER, BOOLEAN, JSON_NULL } tag;
+
+    using ObjectType = std::unordered_map<std::string, JSONValue*>;
+    using ArrayType = std::vector<JSONValue*>;
+    using StringType = std::string;
+    using NumberType = double;
+    using BooleanType = bool;
+    using KeyType = std::string;
+
+    union {
```

Comment:
I suggest encapsulating those as private and providing checking accessors given the trickiness of dealing with unions.